x/incentive/keeper/keeper.go

```diff
@@ -175,6 +176,12 @@ func (k Keeper) UpdateUncommittedTokens(ctx sdk.Context, epochIdentifier string,
 			// Calculate new uncommitted Eden tokens for LP, staker, and Eden token holders
 			newUncommittedEdenTokens := k.CalculateNewUncommittedEdenTokens(ctx, delegatedAmt, commitments, edenAmountPerEpoch)
 
+			// Give commission to validators
+			totalCommissionGiven := k.GiveCommissionToValidators(ctx, creator, delegatedAmt, newUncommittedEdenTokens)
```

Comment:
@kenta-elys  The validator objects have these fields that you may be able to use:

```
// tokens define the delegated tokens (incl. self-delegation).
    Tokens github_com_cosmos_cosmos_sdk_types.Int `protobuf:"bytes,5,opt,name=tokens,proto3,customtype=github.com/cosmos/cosmos-sdk/types.Int" json:"tokens"`
    // delegator_shares defines total shares issued to a validator's delegators.
    DelegatorShares github_com_cosmos_cosmos_sdk_types.Dec `protobuf:"bytes,6,opt,name=delegator_shares,json=delegatorShares,proto3,customtype=github.com/cosmos/cosmos-sdk/types.Dec" json:"delegator_shares"`
```

Reply:
Let me take an example.

Kenta delegated elys to 3 validators - A, B, C among many validators
A has voting power 25 and commission rate 5
B has voting power 15 and commission rate 6
C has voting power 12 and commission rate 7

Kenta awarded 100 Eden at certain epoch according to his total staking share.
If commission rates are all the same and they are 5 %, we can simply get 5 Eden for all 3 validators and distribute 5 according to their voting power.

But commission rate of each validator is different.
So I first calculated the contribution of each validator to 100 according to his share. (In case A, 25/(25+15+12)*100).
And we multiply his commission rate 5 %, and give it to the validator A, It goes the same to the rest of the validators.

That's why I had to iterate delegations and calculate each contribution and get commission amount.

If this make sense.

Reply:
Ah okay, yeah this makes sense.